src/rpc_client/eth_rpc/mod.rs

```diff
@@ -0,0 +1,851 @@
+    }
+}
+
+impl Debug for FixedSizeData {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:x}", self)
+    }
+}
+
+impl Display for FixedSizeData {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:x}", self)
+    }
+}
+
+impl LowerHex for FixedSizeData {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{}", hex::encode(self.0))
+    }
+}
+
+impl UpperHex for FixedSizeData {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{}", hex::encode_upper(self.0))
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+pub enum SendRawTransactionResult {
+    Ok,
+    InsufficientFunds,
+    NonceTooLow,
+    NonceTooHigh,
+}
+
+impl HttpResponsePayload for SendRawTransactionResult {
+    fn response_transform() -> Option<ResponseTransform> {
+        Some(ResponseTransform::SendRawTransaction)
+    }
+}
+
+#[derive(Clone, Copy, Deserialize, Serialize, PartialEq, Eq, Hash, Ord, PartialOrd)]
+pub struct Hash(#[serde(with = "ic_ethereum_types::serde_data")] pub [u8; 32]);
+
+impl Debug for Hash {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:x}", self)
+    }
+}
+
+impl Display for Hash {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:x}", self)
+    }
+}
+
+impl LowerHex for Hash {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{}", hex::encode(self.0))
+    }
+}
+
+impl UpperHex for Hash {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{}", hex::encode_upper(self.0))
+    }
+}
+
+impl std::str::FromStr for Hash {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if !s.starts_with("0x") {
+            return Err("Ethereum hash doesn't start with 0x".to_string());
+        }
+        let mut bytes = [0u8; 32];
+        hex::decode_to_slice(&s[2..], &mut bytes)
+            .map_err(|e| format!("failed to decode hash from hex: {}", e))?;
+        Ok(Self(bytes))
+    }
+}
+
+impl HttpResponsePayload for Hash {}
+
+/// Block tags.
+/// See <https://ethereum.org/en/developers/docs/apis/json-rpc/#default-block>
+#[derive(Debug, Default, Copy, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum BlockTag {
+    /// The latest mined block.
+    #[default]
+    #[serde(rename = "latest")]
+    Latest,
+    /// The latest safe head block.
+    /// See
+    /// <https://www.alchemy.com/overviews/ethereum-commitment-levels#what-are-ethereum-commitment-levels>
+    #[serde(rename = "safe")]
+    Safe,
+    /// The latest finalized block.
+    /// See
+    /// <https://www.alchemy.com/overviews/ethereum-commitment-levels#what-are-ethereum-commitment-levels>
+    #[serde(rename = "finalized")]
+    Finalized,
+    #[serde(rename = "earliest")]
+    Earliest,
+    #[serde(rename = "pending")]
+    Pending,
```

Comment:
Why aren't there any comments for these two fields?

Reply:
Changed [here](https://github.com/internet-computer-protocol/evm-rpc-canister/pull/279/commits/1c9396176a0d17e9791af6deb401b1d9dccd3f51)